android/libraries/rib-base/src/test/java/com/badoo/ribs/core/NodeTest.kt

```diff
@@ -415,6 +431,142 @@ class NodeTest {
         verify(childViewFactory, never()).invoke(parentViewGroup)
     }
 
+    @Test
+    fun `attachChildNode() passes on current lifecycle to direct children - INITIALIZED`() {
+        // by default it's not started, should be on INITIALIZED
+
+        val child = TestNode(
+            savedInstanceState = null,
+            identifier = mock(),
+            viewFactory = mock<TestViewFactory>()
+        )
+        node.attachChildNode(child)
+
+        assertEquals(Lifecycle.State.CREATED, child.externalLifecycleRegistry.currentState)
```

Comment:
is it possible to check behavior using only public interface (what is available for clients) instead of asserting on an internal field?

Reply:
I don't see a quick fix now, but certainly this could be improved (also on other internal fields). I see the bugfix critical enough that I'll move this forward now.